extensions-contrib/kubernetes-overlord-extensions/README.md

```diff
@@ -0,0 +1,222 @@
+    You can override things like labels, environment variables, resources, annotation, or even the base image with this template.
+
+    Required overlord runtime properties
+    ```
+    druid.indexer.runner.k8s.adapter.type=customTemplateAdapter
+    druid.indexer.runner.k8s.podTemplate.base=/opt/druid/conf/druid/cluster/pod-template.yaml` (location of pod template on overlord)
+    ```
+
+    The following PodTemplate is an example that uses the regular druid docker image.
+    ```
+    apiVersion: "v1"
+    kind: "PodTemplate"
+    template:
+      metadata:
+        annotations:
+          custom-annotation: "hello"
+        labels:
+          custom-label: "hello"
+      spec:
+        affinity: {}
+        containers:
+        - command:
+            - sh
+            - -c
+            - |
+              /peon.sh /druid/data 1
+          env:
+          - name: CUSTOM_ENV_VARIABLE
+            value: "hello"
+          image: apache/druid:26.0.0
+          name: main
+          ports:
+          - containerPort: 8091
+            name: druid-tls-port
+            protocol: TCP
+          - containerPort: 8100
+            name: druid-port
+            protocol: TCP
+          resources:
+            limits:
+              cpu: "1"
+              memory: 2400M
+            requests:
+              cpu: "1"
+              memory: 2400M
+          volumeMounts:
+          - mountPath: /opt/druid/conf/druid/cluster/master/coordinator-overlord # runtime props are still mounted in this location because that's where peon.sh looks for configs
+            name: nodetype-config-volume
+            readOnly: true
+          - mountPath: /druid/data
+            name: data-volume
+          - mountPath: /druid/deepstorage
+            name: deepstorage-volume
+        restartPolicy: "Never"
+        securityContext:
+          fsGroup: 1000
+          runAsGroup: 1000
+          runAsUser: 1000
+        tolerations:
+        - effect: NoExecute
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+          tolerationSeconds: 300
+        - effect: NoExecute
+          key: node.kubernetes.io/unreachable
+          operator: Exists
+          tolerationSeconds: 300
+        volumes:
+        - configMap:
+            defaultMode: 420
+            name: druid-tiny-cluster-peons-config
+          name: nodetype-config-volume
+        - emptyDir: {}
+          name: data-volume
+        - emptyDir: {}
+          name: deepstorage-volume
+    ```
+    
+    Runtime properties that need to be passed to the Job's peon process
+    
+    ```
+    druid.port=8100 (what port the peon should run on)
+    druid.peon.mode=remote
+    druid.service=druid/peon (for metrics reporting)
+    druid.indexer.task.baseTaskDir=/druid/data (this should match the argument to the ./peon.sh run command in the PodTemplate)
+    druid.indexer.runner.type=k8s
+    druid.indexer.task.encapsulatedTask=true
```

Comment:
I think you can get rid of this block and move the comments from here down to lines 214 -220.

Reply:
i wanted to be clear that those properties are required, (the peon will fail without them), hopefully we can remove some of them in the future since a lot of them are not necesary